pkg/cloud/gcp/gcp.go

```diff
@@ -46,24 +62,50 @@ func NewGCPProvider(
 		nattPort = helpers.SubmarinerNatTPort
 	}
 
+	if instanceType != "" {
+		instanceType = gwInstanceType
+	}
+
+	if gateways < 1 {
+		return nil, fmt.Errorf("the count of gateways is less than 1")
+	}
+
 	if nattDiscoveryPort == 0 {
 		nattDiscoveryPort = helpers.SubmarinerNatTDiscoveryPort
 	}
 
-	gcpClient, err := client.NewClient(kubeClient, clusterName, credentialsSecretName)
+	projectId, gcpClient, err := newClient(hubKubeClient, clusterName, credentialsSecretName)
+	if err != nil {
+		klog.Errorf("Unable to retrieve the gcpclient :%v", err)
+		return nil, err
+	}
+
+	cloudPrepare := cloudpreparegcp.NewCloud(projectId, infraId, region, gcpClient)
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This isn’t needed, `err` hasn’t changed since the last check.

Reply:
Done